ai: merge in dev and copy over code from old prototype

elbotho opened the pull request:
based on #184 but with "development" changes merged in

hugotiburtino committed:
feat(ai): add endpoint for ai content generation

hugotiburtino committed:
feat(ai): respond to user in case of error while executing prompt

elbotho committed:
Merge branch 'development' into ai

elbotho committed:
add endpoints for exerimental ai features

hugotiburtino commented on src/backend/ai-route-handlers.ts:
```diff
@@ -0,0 +1,146 @@
+const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY })
+
+export async function generateContent(req: Request, res: Response) {
+  if (config.ENVIRONMENT !== 'staging') {
```
```suggestion
  if (config.ENVIRONMENT === 'production') {
```
In order to test it already in the dev environment.
Repeat in the other file.

hugotiburtino reviewed (approved):
Thanks for taking this over and pushing it forward.
I'm in favor of testing it.
We could just make a little change (the one I suggested) in order to test it already in the dev env.

hugotiburtino committed:
refactor(ai): allow it also in dev env

hugotiburtino committed:
refactor(dotenv): remove documentation env var

At the end of the day, we decided not to document all env vars that
are important for other envs besides local, because it would be too
verbose.

hugotiburtino committed:
Merge branch 'development' into ai-experimental-endpoints

hugotiburtino committed:
chore(ci): add env var for openai key
